logbook/types.go

```diff
@@ -0,0 +1,85 @@
+package logbook
```

Comment:
this is _fantastic_. Feels like what the logbook package should have been doing all along